.github/workflows/tests.yml

```diff
@@ -62,11 +62,18 @@ jobs:
             python-version: "3.10"
             allow-failure: false
             test-case: check
-          # remote test
+          # remote test (note that network tests are skipped since the remote server won't have network mode enabled)
           - os: ubuntu-latest
             python-version: "3.11"
             allow-failure: true
             test-case: start test-remote
+          # remote network tests using a remote server with network mode enabled
+          - os: ubuntu-latest
+            python-version: "3.12"
+            allow-failure: true
+            test-case: start test  # the tests that are run are limited by the PYTEST_ADDOPTS in test-option below
+            test-option: >- 
+              PYTEST_ADDOPTS='-m "remote and network"' MAGPIE_NETWORK_CREATE_MISSING_PEM_FILE=True MAGPIE_NETWORK_ENABLED=on MAGPIE_NETWORK_INSTANCE_NAME=example
```

Comment:
While this works functionality-wise, the CI displays it not that nicely. It might be worth having a dedicated `test-remote-network` target that does these steps to facilitate running it locally and understand the CI matrix operations more easily.

<img height="400" alt="Image" src="https://github.com/user-attachments/assets/05c2b39c-ad00-4c16-a150-b2290ecae60a" />

<img width="932" height="389" alt="Image" src="https://github.com/user-attachments/assets/fca5c281-4992-4c67-9a1f-877f2b20cba0" />